File: task1.cpp

using namespace std;

bool isEven(int value){
    return value % 2 == 0;
}

bool isEvenBinary(int value){
    return (value & 1) == 0;
}


int main() {
    int n;
    string st;
    cin >> st;
    char last_digit = st[st.size() - 1];
    cout << isEven(stoi(string({last_digit}))) << "\n";
    cout << isEvenBinary(stoi(st));
    return 0;
}

File: task1.py
def is_even_binary(value):
    return bool(value & 1)


def is_even(value):
    return int(str(value)[-1]) % 2 == 0


if __name__ == '__main__':
    value = int(input())
    print(is_even(value))
    print(is_even_binary(value))
